torch/csrc/autograd/python_variable.cpp

```diff
@@ -485,6 +489,22 @@ namespace torch { namespace autograd {
 extern PyMethodDef variable_methods[];
 extern void initTorchFunctions(PyObject *module);
 
+void initTensorImplConversion(PyObject* module) {
+  auto m = py::handle(module).cast<py::module>();
+  m.def("_wrap_tensor_impl", [](void* ptr) {
+    auto p =
+        c10::intrusive_ptr<c10::TensorImpl, at::UndefinedTensorImpl>::reclaim(
+            static_cast<c10::TensorImpl*>(ptr));
```

Comment:
I'd feel better if we had an assert here that the TensorImpl pointer in question was not a variable.

Reply:
In particular, I don't think `wrap_tensor_impl` will enforce the invariant for you.